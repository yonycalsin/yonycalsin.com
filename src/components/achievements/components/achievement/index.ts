export { default as Achievement } from './achievement'
