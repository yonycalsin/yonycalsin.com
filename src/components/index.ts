export * from './achievements'
export * from './command-bar'
export * from './external-anchor'
export * from './float-socials'
export * from './illustrations'
export * from './mdx-components'
export * from './meta'
export * from './motion'
export * from './night-mode-button'
export * from './page-transition'
export * from './recommendations'
export * from './section'
export * from './work-item'
