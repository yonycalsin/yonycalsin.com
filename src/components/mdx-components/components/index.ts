export * from './anchor'
export * from './linked-heading'
export * from './table'
