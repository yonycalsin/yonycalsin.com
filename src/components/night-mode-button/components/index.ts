export * from './moon-icon'
export * from './sun-icon'
