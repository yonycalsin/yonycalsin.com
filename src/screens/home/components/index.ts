export * from './featured-achievements'
export * from './featured-recommendations'
export * from './introduction-section'
export * from './pinned-projects'
