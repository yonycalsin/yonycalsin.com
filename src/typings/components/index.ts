export * from './achievements'
export * from './anchor'
export * from './badge'
export * from './blockquote'
export * from './breadcrumb'
export * from './button'
export * from './divider'
export * from './external-anchor'
export * from './heading'
export * from './icon'
export * from './illustrations'
export * from './list'
export * from './loaders'
export * from './meta'
export * from './night-mode-button'
export * from './page-transaction'
export * from './recommendations'
export * from './section'
export * from './work-item'
