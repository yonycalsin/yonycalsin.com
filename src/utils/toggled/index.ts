export { default as buildFeatures } from './build-features'
