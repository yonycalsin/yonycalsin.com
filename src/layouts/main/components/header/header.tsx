import isotipo from 'assets/brand/images/isotipo.png'

import * as React from 'react'
import { Box, Container, HStack, Link } from '@chakra-ui/react'
import Image from 'next/image'
import RouterLink from 'next/link'
import { Flag } from 'toggled'

import { Features } from 'utils/constants'

const Header = () => {
  return (
    <Box
      w="full"
      top="0"
      zIndex="banner"
      left="0"
      boxShadow="xs"
      _dark={{
        boxShadow: 'lg',
      }}
    >
      <Container
        maxW="container.md"
        py={{
          base: '5',
          md: '8',
        }}
        display="flex"
        justifyContent="space-between"
        alignItems="center"
      >
        <RouterLink href="/" passHref>
          <Link display="contents">
            <Image src={isotipo} layout="intrinsic" width={32} height={32} priority alt="" />
          </Link>
        </RouterLink>
        <HStack spacing="4">
          <RouterLink href="/projects" passHref>
            <Link color="primary.500" fontWeight="bold">
              Projects
            </Link>
          </RouterLink>
          <RouterLink href="/me" passHref>
            <Link color="primary.500" fontWeight="bold">
              About Me
            </Link>
          </RouterLink>
          <Flag flagQuery={Features.FAQ}>
            <RouterLink href="/faq" passHref>
              <Link color="primary.500" fontWeight="bold">
                FAQ
              </Link>
            </RouterLink>
          </Flag>
          <Flag flagQuery={Features.USES}>
            <RouterLink href="/uses" passHref>
              <Link
                color="primary.500"
                fontWeight="bold"
                display={{
                  base: 'none',
                  md: 'block',
                }}
              >
                Uses
              </Link>
            </RouterLink>
          </Flag>
          <Flag flagQuery={Features.SNIPPETS}>
            <RouterLink href="/blog/categories/snippet" passHref>
              <Link
                color="primary.500"
                fontWeight="bold"
                display={{
                  base: 'none',
                  md: 'block',
                }}
              >
                Snippets
              </Link>
            </RouterLink>
          </Flag>
        </HStack>
      </Container>
    </Box>
  )
}

export default Header
