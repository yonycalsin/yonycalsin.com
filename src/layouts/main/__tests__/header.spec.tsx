import { render } from '@testing-library/react'
import { FeatureProvider } from 'toggled'

import { Header } from '../components/header'

describe('Layouts / Main', () => {
  it('renders a header', () => {
    const view = render(
      <FeatureProvider features={[]}>
        <Header />
      </FeatureProvider>,
    )

    expect(view.baseElement).toMatchSnapshot()
  })
})
