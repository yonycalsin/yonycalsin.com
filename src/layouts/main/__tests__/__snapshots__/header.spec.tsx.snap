// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout/Header renders a header 1`] = `
.emotion-0 {
  width: full;
  top: 0px;
  z-index: banner;
  left: 0px;
  box-shadow: xs;
}

.chakra-ui-dark .emotion-0:not([data-theme]),
[data-theme=dark] .emotion-0:not([data-theme]),
.emotion-0[data-theme=dark] {
  box-shadow: lg;
}

.emotion-1 {
  max-width: container.md;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 py {
  base: 5;
  md: 8;
}

.emotion-2 {
  display: contents;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-3>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 4px;
  margin-inline-start: 4px;
}

.emotion-4 {
  color: primary.500;
  font-weight: bold;
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      <div
        class="chakra-container emotion-1"
      >
        <a
          class="chakra-link emotion-2"
          href="/"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt=""
              data-nimg="intrinsic"
              decoding="async"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </span>
        </a>
        <div
          class="chakra-stack emotion-3"
        >
          <a
            class="chakra-link emotion-4"
            href="/projects"
          >
            Projects
          </a>
          <a
            class="chakra-link emotion-4"
            href="/me"
          >
            About Me
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
